backend/fms_core/resources/sample.py

```diff
@@ -118,6 +120,7 @@ def before_import(self, dataset, using_transactions, dry_run, **kwargs):
 
     def import_obj(self, obj, data, dry_run):
         errors = {}
+        self.project = {}
```

Comment:
This is a dangerous initialization ... you should use None. Setting self.project with {} make a dictionary out of it ... self.project is a Field() normally... You are changing the type which may have unintended consequences.

Reply:
Agreed. In your Sample resource, the field defined as being 'project = Field(attribute='projects', column_name='Project', widget=ManyToManyWidget(Project))' can actually be referenced with self.project since it is a class attribute.

Re-initiating self.project = {} in import_obj can be dangerous here. You could simply rename to variable to something like self.project_obj for example to make a clear distinction